### ISSUE TITLE
Add way to auto-format VCS changed files without staging them

### ISSUE THREAD

**Issue by ZakTaccardi** [JLLeitschuh/ktlint-gradle#331]
I would like the effects of a `ktlintFormat` precommit hook, but without staging and committing automatically. It would be good to see what the autoformatting caught before committing it

**Comment by Tapchicoma** [comment]
Have you considered using git hook from `addKtlintCheckGitPreCommitHook` task?

**Comment by ZakTaccardi** [comment]
> Have you considered using git hook from addKtlintCheckGitPreCommitHook task?

Yes - currently using it, but I would prefer to have an "in-between" the `check` and `format` hooks, where it formats the VCS changed files (without staging/adding them) and fails the commit in case of a violation. This allows the developer to review what `ktlintFormat` has modified quickly.

My understanding is that the downside of `addKtlintFormatGitPreCommitHook` is that it will format and commit the formatted changes automatically - and there is no developer review of what `ktlintFormat` has modified.

**Comment by Tapchicoma** [comment]
How often have you changed something that was modified by ktlint format? :thinking: 

Plugin should provide only generic hooks, but not some user specific one (for example someone may want to have check/format in `pre-push` hook). Though for such cases plugin could expose some methods/tasks allowing user to add their own specific git hook.
